spring-shell-standard/src/main/java/org/springframework/shell/standard/StandardMethodTargetRegistrar.java

```diff
@@ -74,11 +79,75 @@ public void register(ConfigurableCommandRegistry registry) {
 				}
 				String group = getOrInferGroup(method);
 				for (String key : keys) {
+					log.debug("Registering with keys='{}' key='{}'", keys, key);
 					Supplier<Availability> availabilityIndicator = findAvailabilityIndicator(keys, bean, method);
-					MethodTarget target = new MethodTarget(method, bean, new Command.Help(shellMapping.value(), group),
-							availabilityIndicator, shellMapping.interactionMode());
-					registry.register(key, target);
-					commands.put(key, target);
+
+					Builder builder = CommandRegistration.builder()
+						.command(key)
+						.group(group)
+						.help(shellMapping.value())
+						.interactionMode(shellMapping.interactionMode())
+						.availability(availabilityIndicator);
+
+					InvocableHandlerMethod ihm = new InvocableHandlerMethod(bean, method);
```

Comment:
Can you leave a comment here describing what all this is doing? I got lost on this one. :)